### PR TITLE
Fix parsing 'A' large-arc and sweep flags when not separated by space or comma

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes https://github.com/googlefonts/picosvg/issues/171

### CODE REVIEW

File: src/picosvg/svg_path_iter.py [hunk 1]
```diff
@@ -13,16 +13,69 @@
...
+    while raw_args:
+        arg = raw_args.pop()
+        # modulo to wrap around
+        converter, regex = arg_types[i % n]
+        m = regex.match(arg)
+        if not m:
+            raise ValueError(f"Invalid argument #{i} for '{cmd}': {arg!r}")
+
+        start, end = m.span()
+        yield converter(arg[start:end])
```
Comment (rsheeter) [inline src/picosvg/svg_path_iter.py:71]: Why not capture a group?

File: src/picosvg/svg_path_iter.py [hunk 2]
```diff
@@ -13,16 +13,69 @@
...
 _IMPLICIT_REPEAT_CMD = {"m": "l", "M": "L"}
 
 
+def _parse_args(cmd: str, args: str) -> Generator[float, None, None]:
+    raw_args = [s for s in _SEPARATOR_RE.split(args) if s]
+    if not raw_args:
+        return
+
+    # reverse so we can pop()/append() like a stack, quicker than pop(0)/insert(0)
+    raw_args.reverse()
```
Comment (rsheeter) [inline src/picosvg/svg_path_iter.py:53]: Surely the pop(0) overhead on such tiny lists is immaterial?

Comment (anthrotype) [reply src/picosvg/svg_path_iter.py:53]: maybe, I don't mind removing it, though it seemed uncontroversial to me. popping/inserting from the beginning of a list always reallocates memory

File: src/picosvg/svg_path_iter.py (hunk 1 above)
Comment (anthrotype) [reply src/picosvg/svg_path_iter.py:71]: I need the span().end to so that I can check if there's any remainder, so I might as well use it?

Comment (anthrotype) [reply src/picosvg/svg_path_iter.py:71]: also I want the whole FLOAT_RE match, not any particular part of it, so a group doesn't seem very useful here

**Review by rsheeter — APPROVED** [review]
Feels like we're tiptoing toward writing a proper parser :D

**Comment by anthrotype** [comment]
> Feels like we're tiptoing toward writing a proper parser :D

unless SVG 3.0 makes d strings even crazier, I think with this we cover everything (although I might have to add more tests to convince myself of that). For now, I think this is good enough

File: src/picosvg/svg_path_iter.py [hunk 3]
```diff
@@ -13,16 +13,69 @@
...
+
+    if cmd.upper() == "A":
+        arg_types = _ARC_ARGUMENT_TYPES
+    else:
+        arg_types = ((float, _FLOAT_RE),)
+    n = len(arg_types)
+
+    i = 0
+    while raw_args:
```
Comment (rsheeter) [inline src/picosvg/svg_path_iter.py:62]: Optional alternative: while i < len(raw_args), instead of pop just read raw_args[i].

Comment (anthrotype) [reply src/picosvg/svg_path_iter.py:62]: I don't think that will work actually. `i` counts the argument number (there are 7 in an arc), raw_args are a list of str (split by comma or space), the length of that list is not necessarily == number of logical argument, as more arguments may be coalesced into a single string (which is why I match the beginning of the arg string, then push the tail back to raw_args so I can parse at the next iteration). I think it's good like this.

Comment (anthrotype) [reply src/picosvg/svg_path_iter.py:62]: ok if I use two distinct iteration variables, `i` and `j` I can do without the pop and push, lemme try

**Pull request merged**